### PR TITLE
🥒 Enable javascript in cukes

### PR DESCRIPTION
**Pull request by josemigallas**
**NOTE**: only add `@javascript` to scenarios logging in to a provider

Javascript is disabled for most cucumbers and it creates some false positives. For instance, some forms have HTML5 validations that prevent them from being submitted, but it won't work unless `@javscript` tag is used.

On the other hand the login page has a no-javascript alternative (that nobody uses because porta needs JS to work) that will be removed in #3316, and will be a React rendered component, so @javascript has to be enabled.

### CODE REVIEW

**Review by lvillen — COMMENTED** [review]
Some tests are failing, but overall looks good, when tests are green count with my approval 👍

File: app/inputs/price_input.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def to_html
   end
 
   def input_html_options
-    super.merge({ value: value, min: 0, step: 0.01 })
+    super.merge({ value: value, step: 0.001 })
```
Comment (akostadinov) [inline app/inputs/price_input.rb:19]: why this change?

Comment (josemigallas) [reply app/inputs/price_input.rb:19]: _fixes price input so that tests pass_

According to tests, this is what price input should accept.

File: features/support/invoices_helpers.rb [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+
 module InvoicesHelpers
   def create_invoice(buyer, date, opts = {})
     options = opts.reverse_merge(buyer_account: buyer, period: Month.new(date), creation_type: :background)
     buyer.provider_account.billing_strategy.create_invoice!(options)
   end
 
   def assert_secure_invoice_pdf_url(url, invoice)
-    assert_equal invoice.pdf.expiring_url, url
+    assert url.ends_with?(invoice.pdf.expiring_url)
```
Comment (akostadinov) [inline features/support/invoices_helpers.rb:10]: why no match anymore?

File: features/step_definitions/security/generic_steps.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
+# frozen_string_literal: true
+
 Then /^I should be denied the access$/ do
-  # assert has_content?('Access Denied')
-  assert_equal 403, page.status_code
+  assert_content 'Access Denied'
```
Comment (akostadinov) [inline features/step_definitions/security/generic_steps.rb:4]: Also could you explain this?

File: features/step_definitions/provider_steps.rb [hunk 4]
```diff
@@ -147,6 +147,7 @@ def setup_provider(login)
   step 'I log in as provider "foo.3scale.localhost"' if login
 
   @provider = Account.find_by_domain!('foo.3scale.localhost')
+  @service ||= @provider.default_service
```
Comment (akostadinov) [inline features/step_definitions/provider_steps.rb:150]: Can service be already set? Maybe just `=`?

File: features/step_definitions/menu_steps.rb [hunk 5]
```diff
@@ -1,21 +1,23 @@
...
-  items.raw.each do |item|
-    within '#mainmenu' do
-      assert has_css? 'li', :text => item[0]
-    end
+  within '#mainmenu' do
+    assert_equal items.raw.flatten, find_all('#mainmenu .pf-c-nav__list > .pf-c-nav__item > .pf-c-nav__link').map(&:text)
   end
 end
 
-Then /^I should not see menu items$/ do |items|
```
Comment (akostadinov) [inline features/step_definitions/menu_steps.rb:9]: This step not needed anymore?

File: features/step_definitions/finance/settings_steps.rb [hunk 6]
```diff
@@ -19,7 +19,6 @@
 
 Then "they should not be able to review the charging and gateway billing settings" do
   visit admin_finance_settings_path
-  assert_equal 403, status_code
```
Comment (akostadinov) [inline features/step_definitions/finance/settings_steps.rb:22]: this stopped working somehow?

File: features/step_definitions/apicast_steps.rb [hunk 7]
```diff
@@ -13,5 +13,5 @@
 
 Given(/^the default proxy does not use apicast configuration driven$/) do
   proxy = @provider.default_service.proxy
-  proxy.update!(apicast_configuration_driven: false, sandbox_endpoint: 'https://api-2.staging.apicast.io:4443')
+  proxy.update!(apicast_configuration_driven: false, sandbox_endpoint: 'https://api-2.staging.apicast.io:4443', endpoint: 'https://api-2.staging.io:4443')
```
Comment (akostadinov) [inline features/step_definitions/apicast_steps.rb:16]: would be interesting to know why this change was needed

**Review by akostadinov — COMMENTED** [review]
Looks nice. besides 2 nitpicks, most comments are just asking to explain. Thank you!

**Review by lvillen — APPROVED** [review]

File: features/step_definitions/finance/settings_steps.rb (hunk 6 above)
Comment (josemigallas) [reply features/step_definitions/finance/settings_steps.rb:22]: Yes, `status_code` is not meant for browser tests so it's not available anymore.

File: features/step_definitions/menu_steps.rb (hunk 5 above)
Comment (josemigallas) [reply features/step_definitions/menu_steps.rb:9]: I had to fix the other step `I should see menu items` and now it will fail unless _all_ menu items match, so it already assert missing menu items.

Comment (akostadinov) [reply features/step_definitions/menu_steps.rb:9]: maybe call the other step `I should see only menu items`

File: features/step_definitions/provider_steps.rb (hunk 4 above)
Comment (josemigallas) [reply features/step_definitions/provider_steps.rb:150]: Good point. I don't think it should be set here actually. This `@service` is reserved for steps that refer to a service. This one is about the provider.

File: features/step_definitions/security/generic_steps.rb (hunk 3 above)
Comment (josemigallas) [reply features/step_definitions/security/generic_steps.rb:4]: Selenium does not support `status_code`: https://github.com/seleniumhq/selenium-google-code-issue-archive/issues/141

File: features/api/services/api_docs.feature [hunk 8]
```diff
@@ -34,19 +33,19 @@ Feature: ActiveDocs
       | swagger version |
       | Swagger 1.2     |
       | Swagger 2       |
-      # | OAS 3.0         | Feature not implemented |
+  # | OAS 3.0         | Feature not implemented |
```
[the remark this answers is not in the document]
Comment (josemigallas) [reply features/api/services/api_docs.feature:36]: It seems to be a bug in the formatter, it works all right with commented table rows in the middle. I'll fix it manually.

File: features/step_definitions/apicast_steps.rb (hunk 7 above)
Comment (josemigallas) [reply features/step_definitions/apicast_steps.rb:16]: Indeed... the error was "Base URL can't be blank" but I wonder why the validation did not work before.

Comment (akostadinov) [reply features/step_definitions/apicast_steps.rb:16]: possible that in the past that attribute was already set by previous steps?

Comment (josemigallas) [reply features/step_definitions/apicast_steps.rb:16]: I don't think any step is being removed by this PR 🤔

File: features/step_definitions/menu_steps.rb (hunk 5 above)
Comment (josemigallas) [reply features/step_definitions/menu_steps.rb:9]: That's `I should see menu items` 😄
